torch/csrc/jit/codegen/cuda/index_compute.h

```diff
@@ -59,6 +59,22 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+struct ReferencePredicate {
```

Comment:
Can clean this up, but may reuse the struct to return information to unswitch predicate.